### PR TITLE
Refund any prize message balance once it has ended

### PR DESCRIPTION
**Pull request by megrogan**
Still needs an integration test(s)

### CODE REVIEW

File: backend/canisters/community/impl/src/timer_job_types.rs [hunk 1]
```diff
@@ -130,3 +149,43 @@ impl Job for MarkGroupImportCompleteJob {
...
+        ic_cdk::spawn(make_transfer(pending, sender));
+
+        async fn make_transfer(pending_transaction: PendingCryptoTransaction, sender: CanisterId) {
+            if let Err(error) = process_transaction(pending_transaction.clone(), sender).await {
+                error!(?error, "Transaction failed");
+                mutate_state(|state| {
+                    let now = state.env.now();
+                    state.data.timer_jobs.enqueue_job(
+                        TimerJob::MakeTransfer(MakeTransferJob { pending_transaction }),
+                        now + HOUR_IN_MS,
```
Comment (hpeebles) [inline backend/canisters/community/impl/src/timer_job_types.rs:184]: I reckon an hour is too long. The main cause of failure would be the subnet being upgraded which only lasts a few minutes. So maybe retry after 1 min?

Comment (megrogan) [reply backend/canisters/community/impl/src/timer_job_types.rs]: changed

File: backend/libraries/chat_events/src/chat_events.rs [hunk 2]
```diff
@@ -407,6 +408,42 @@ impl ChatEvents {
...
+        use ClosePrizeResult::*;
+        let now_ms = now_nanos / 1_000_000;
+        if let Some((message, _)) =
+            self.message_internal_mut(EventIndex::default(), thread_root_message_index, message_index.into(), now_ms)
+        {
+            if let MessageContentInternal::Prize(p) = &mut message.content {
+                match &p.transaction {
+                    CryptoTransaction::Pending(_) => return FundTransferPending,
+                    CryptoTransaction::Completed(t) => {
+                        let unclaimed = p.prizes_remaining.iter().map(|t| t.e8s() as u128).sum::<u128>();
```
Comment (hpeebles) [inline backend/libraries/chat_events/src/chat_events.rs:426]: If there are multiple prizes remaining we should also refund multiple fees.

File: backend/integration_tests/src/communities/send_message_tests.rs [hunk 3]
```diff
@@ -1,14 +1,17 @@
+use crate::client::icrc1::happy_path;
```
Comment (hpeebles) [inline backend/integration_tests/src/communities/send_message_tests.rs:1]: Seems strange for `happy_path` within a communities test to be the `icrc1::happy_path`

**Review by hpeebles — APPROVED** [review]

**Pull request merged**